pagopa/io-app
# chore: [IOBP-168,IOBP-167] Toasts messages and native alert for delete payment method confirmation

# ⚠️ This PR depends on #4867 ⚠️

## Short description
This PR adds:
- Toast notification feedback (in case of error or success) when enabling or disabling the "Pagamenti in app" toggle;
- Native alert to confirm the deletion of the payment method;

## Design reference
[FIGMA](https://www.figma.com/file/f7jOjE3RokprMfb7c9NCVQ/Dettaglio-metodo-di-pagamento?type=design&node-id=1205-19503&mode=design&t=FoEVzT6goyKKIWNe-4)

## List of changes proposed in this pull request
- Added new translated text into locale files;
- Added into `BasePaymentMethodScreen` a function `onDeleteMethods` in order to show the native Alert to confirm or not the deletion;
- Added into `ts/sagas/wallet/pagopaApis.ts`a success toast showed when the updatePaymentStatus request is executed successfully;
- Replaced into `PaymentStatusSwitch` the old toast notification handler with the new one `IOToast`;
- Removed `RemovePaymentMethod` component and its hook since it is not used anymore;

## How to test
- Go into a payment method detail
- Press on "Elimina questo metodo" button and it should appear a native Alert to confirm or not the choose;
- Toggle the "Pagamenti in app" switch and should show a success/error toast feedback message.

## Preview


https://github.com/pagopa/io-app/assets/34343582/4286f296-e55b-48eb-87ed-1a4c4f26e1f2
